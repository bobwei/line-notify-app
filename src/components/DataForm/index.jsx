import React from 'react';
import { Row, Col, Alert, Form } from 'react-bootstrap';
import { Field } from 'redux-form';
import R from 'ramda';
import mapProps from 'recompose/mapProps';

import FieldGroup from '../FieldGroup';
import Button from '../Button';
import styles from './index.scss';

const DataForm = ({
  handleSubmit, submitting, error,
  alert,
  fields,
}) => (
  <Form onSubmit={handleSubmit}>
    {error && React.createElement(alert, { error })}
    {fields.map(field => (
      <Field
        key={field.name}
        component={FieldGroup}
        {...field}
      />
    ))}
    <Row>
      <Col mdOffset={4} md={4}>
        <Button bsStyle="main" className={styles.btnSubmit} type="submit" disabled={submitting} block>
          {(submitting) ? '讀取中...' : '登入'}
        </Button>
      </Col>
    </Row>
  </Form>
);

DataForm.defaultProps = {
  alert: mapProps(({ error }) => ({
    bsStyle: 'warning',
    children: R.ifElse(R.is(String), R.identity, str => JSON.stringify(str))(error),
  }))(Alert),
  fields: [],
};

DataForm.propTypes = {
  /* form props */
  handleSubmit: React.PropTypes.func,
  submitting: React.PropTypes.bool,
  error: React.PropTypes.string,

  /* component props */
  alert: React.PropTypes.func,
  fields: React.PropTypes.arrayOf(React.PropTypes.shape({
    ...FieldGroup.propTypes,
  })),
};

export default DataForm;
