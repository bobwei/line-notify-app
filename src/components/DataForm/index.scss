.btnSubmit {
  margin-top: 30px;
}
