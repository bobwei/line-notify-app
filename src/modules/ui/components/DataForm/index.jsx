import React from 'react';
import { Row, Col, Alert, Form } from 'react-bootstrap';
import { Field } from 'redux-form';
import R from 'ramda';
import mapProps from 'recompose/mapProps';

import FieldGroup from '../FieldGroup';
import styles from './index.scss';
import SubmitButton from '../SubmitButton';

const DataForm = ({
  handleSubmit,
  submitting,
  error,
  alert,
  fields,
  submitButton,
}) =>
  <Form onSubmit={handleSubmit} noValidate>
    {error && React.createElement(alert, { error })}
    {fields.map(field =>
      <Field key={field.name} component={FieldGroup} {...field} />,
    )}
    <Row>
      <Col mdOffset={4} md={4}>
        {React.createElement(submitButton, {
          submitting,
          className: styles.btnSubmit,
          block: true,
        })}
      </Col>
    </Row>
  </Form>;

DataForm.defaultProps = {
  alert: mapProps(({ error }) => ({
    bsStyle: 'warning',
    children: R.ifElse(R.is(String), R.identity, str => JSON.stringify(str))(
      error,
    ),
  }))(Alert),
  fields: [],
  submitButton: SubmitButton,
};

DataForm.propTypes = {
  /* form props */
  handleSubmit: React.PropTypes.func,
  submitting: React.PropTypes.bool,
  error: React.PropTypes.string,

  /* component props */
  alert: React.PropTypes.func,
  fields: React.PropTypes.arrayOf(
    React.PropTypes.shape({
      ...FieldGroup.propTypes,
    }),
  ),
  submitButton: React.PropTypes.func,
};

export default DataForm;
