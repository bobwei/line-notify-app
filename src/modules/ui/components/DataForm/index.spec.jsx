/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import renderer from 'react-test-renderer';

import DataForm from './index';

it('can render form correctly', () => {
  const component = renderer.create(<DataForm />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('can render form with error correctly', () => {
  const component = renderer.create(<DataForm error="login error" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
