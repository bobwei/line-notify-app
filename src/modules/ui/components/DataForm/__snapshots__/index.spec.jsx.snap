// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render form correctly 1`] = `
<form
  className=""
  noValidate={true}
  onSubmit={undefined}
>
  <div
    className="row"
  >
    <div
      className="col-md-4 col-md-offset-4"
    >
      <button
        className="btnSubmit main btn btn-block"
        disabled={false}
        type="submit"
      >
        Save
      </button>
    </div>
  </div>
</form>
`;

exports[`can render form with error correctly 1`] = `
<form
  className=""
  noValidate={true}
  onSubmit={undefined}
>
  <div
    className="alert alert-warning"
    role="alert"
  >
    login error
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-4 col-md-offset-4"
    >
      <button
        className="btnSubmit main btn btn-block"
        disabled={false}
        type="submit"
      >
        Save
      </button>
    </div>
  </div>
</form>
`;
